src/views/Workspace.vue

```diff
@@ -37,19 +36,45 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 </template>
 
 <script>
+import { defineAsyncComponent } from 'vue'
 import { uniqueId } from 'lodash'
 import { toArray } from '@lumino/algorithm'
+import {
+  mdiChartLine,
+  mdiFileDocumentMultipleOutline,
+  mdiFileTree,
+  mdiGraph,
+  mdiTable,
+  mdiTree,
+} from '@mdi/js'
 import { getPageTitle } from '@/utils/index'
 import graphqlMixin from '@/mixins/graphql'
 import subscriptionMixin from '@/mixins/subscription'
 import ViewState from '@/model/ViewState.model'
 import Lumino from '@/components/cylc/workflow/Lumino.vue'
 import Toolbar from '@/components/cylc/workflow/Toolbar.vue'
-import TableView from '@/views/Table.vue'
-import TreeView from '@/views/Tree.vue'
-import GraphView from '@/views/Graph.vue'
-import LogView from '@/views/Log.vue'
-import AnalysisView from '@/views/Analysis.vue'
+
+// Use dynamic async components for lazy loading:
+const TreeView = defineAsyncComponent(() => import('@/views/Tree.vue'))
+const TableView = defineAsyncComponent(() => import('@/views/Table.vue'))
+const GraphView = defineAsyncComponent(() => import('@/views/Graph.vue'))
+const LogView = defineAsyncComponent(() => import('@/views/Log.vue'))
+const AnalysisView = defineAsyncComponent(() => import('@/views/Analysis.vue'))
+const SimpleTreeView = defineAsyncComponent(() => import('@/views/SimpleTree.vue'))
+
+const allViews = [
+  { name: 'Tree', component: TreeView, icon: mdiFileTree },
+  { name: 'Table', component: TableView, icon: mdiTable },
+  { name: 'Graph', component: GraphView, icon: mdiGraph },
+  { name: 'Log', component: LogView, icon: mdiFileDocumentMultipleOutline },
+  { name: 'Analysis', component: AnalysisView, icon: mdiChartLine },
+]
```

Comment:
For the long term we should think about how to make views more "plugable". It would be helpful for this information to remain tethered with the view, perhaps in some kind of metadata / sidecar file that could be imported separately. Would be worth taking a look at how Jupyter Lab make their "views" plugable as they face a similar (but more advanced) problem here.

No need for changes, this is great for now, but something to store in the back of the mind for later work...

Reply:
Added a "pending" issue to preserve this, https://github.com/cylc/cylc-ui/issues/1319